cassandra/cluster.py

```diff
@@ -3364,8 +3369,6 @@ def _on_speculative_execute(self):
         self._timer = None
         if not self._event.is_set():
             if self._time_remaining is not None:
```

Comment:
I see.... you could remove this IF if you just return 0 in _time_remaining I think.